tests/transformers/tests/generation/test_utils.py

```diff
@@ -98,109 +107,82 @@ def _update_default_model_kwargs(self, model_kwargs):
 
     def _get_input_ids_and_config(self, batch_size=2):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
-        input_ids = inputs_dict[self.input_name]
+        # TODO: @raushan or @gante, use `model.main_input_name` as the main input instead of relyinn on `input_ids`
+        input_ids = inputs_dict.pop(self.input_name)[:batch_size, :]
+        inputs_dict.pop("attention_mask", None)
+
+        # we don't want encoder-decoder models to start from filled decoder ids
+        inputs_dict.pop("decoder_input_ids", None)
+        inputs_dict.pop("decoder_attention_mask", None)
 
         # cut to half length & take max batch_size 3
         sequence_length = input_ids.shape[-1] // 2
         input_ids = input_ids[:batch_size, :sequence_length]
 
-        # generate max 3 tokens
-        max_length = input_ids.shape[-1] + 3
+        # we'll set cache use in each test differently
+        inputs_dict.pop("use_cache", None)
+
+        inputs_dict = {
+            k: v[:batch_size, ...]
+            for k, v in inputs_dict.items()
+            if "head_mask" not in k and isinstance(v, torch.Tensor)
+        }
         if config.eos_token_id is not None and config.pad_token_id is None:
             # hack to allow generate for models such as GPT2 as is done in `generate()`
             if isinstance(config.eos_token_id, int):
                 config.eos_token_id = [config.eos_token_id]
             config.pad_token_id = config.eos_token_id[0]
-        # TransfoXL has no attention mask
-        if "transfoxl" in config.__class__.__name__.lower():
-            attention_mask = None
+
+        if self.has_attentions:
+            attention_mask = torch.ones_like(input_ids, dtype=torch.long)
         else:
-            attention_mask = torch.ones_like(input_ids, dtype=torch.long)[:batch_size, :sequence_length]
-
-        return config, input_ids, attention_mask, max_length
-
-    @staticmethod
-    def _get_logits_processor_and_kwargs(
-        input_length,
-        eos_token_id,
-        forced_bos_token_id=None,
-        forced_eos_token_id=None,
-        max_length=None,
-        diversity_penalty=None,
-    ):
-        process_kwargs = {
-            "min_length": input_length + 1 if max_length is None else max_length - 1,
+            attention_mask = None
+
+        # It is important set set the eos_token_id to None to ensure that no sequences
+        # shorter than `max_length` can be generated
+        config.eos_token_id = None
```

Comment:
@malkomes this is confusing again!

Above we are setting `config.eos_token_id = [config.eos_token_id]` and here you are un-setting it `config.eos_token_id = None`?
If it is needed to be set to None, we should remove the upper code and just set `config.pad_token_id` there

Reply:
great catch! that's something that comes from the [upstream version](https://github.com/huggingface/transformers/blob/53fad641cfdb5105e2470bcf3ef17ea8e25cc300/tests/generation/test_utils.py#L121-L131) that I was referring. I'll fix it.